src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -231,4 +247,39 @@ namespace AppInstaller::CLI::Workflow
             trackingCatalog.RecordUninstall(item.Identifier);
         }
     }
+
+    void ReportUninstallerResult::operator()(Execution::Context& context) const
+    {
+        DWORD uninstallResult = context.Get<Execution::Data::OperationReturnCode>();
+        if (uninstallResult != 0)
+        {
+            const auto installedPackageVersion = context.Get<Execution::Data::InstalledPackageVersion>();
+            Logging::Telemetry().LogUninstallerFailure(
+                installedPackageVersion->GetProperty(PackageVersionProperty::Id),
+                installedPackageVersion->GetProperty(PackageVersionProperty::Version),
+                m_uninstallerType,
+                uninstallResult);
+
+            if (m_isHResult)
+            {
+                context.Reporter.Error() << Resource::String::UninstallFailedWithCode << ' ' << GetUserPresentableMessage(uninstallResult) << std::endl;
+            }
+            else
+            {
+                context.Reporter.Error() << Resource::String::UninstallFailedWithCode << ' ' << uninstallResult << std::endl;
+            }
+
+            // Show installer log path if exists
+            if (context.Contains(Execution::Data::LogPath) && std::filesystem::exists(context.Get<Execution::Data::LogPath>()))

```

Comment:
I don't think we have an uninstall log file, but if we ever happen to I suppose this is ready for it.